udev/80-azure-nvme.rules.in

```diff
@@ -1,7 +1,7 @@
 ACTION!="add|change", GOTO="azure_nvme_end"
 SUBSYSTEM!="block", GOTO="azure_nvme_end"
 KERNEL!="nvme*", GOTO="azure_nvme_end"
-ATTR{nsid}!="?*", GOTO="azure_nvme_end"
+ENV{ID_NSID}!="?*", GOTO="azure_nvme_end"
```

Comment:
out of curiosity is there a reason when to use one method over the other?

Reply:
`ATTR{nsid}` is only present on the disk (nvme0n1), so it doesn't match for partitions. You can either use `ATTRS{nsid}` to get matching against parent attributes, or switch to `ENV{ID_NSID}`. Udev itself ships the rule that does:
```
ATTRS{nsid}=="?*",         ENV{ID_NSID}="$attr{nsid}"
``` 
so I'd think that they want people using `ENV{ID_NSID}`. The same applies to `ENV{ID_MODEL}`, it's synthesized from `ATTRS{model}`.

Reply:
thanks! makes sense, I figured there was a good reason :D